gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestMapperTest.java

```diff
@@ -338,399 +341,361 @@ public void When_MappingMedicationRequestWithAuthoredOnValidDateInExtractOnly_Ex
-            .toList();
+        assertThrows(
+                NullPointerException.class,
+                () -> medicationRequestMapper.mapResources(
+                        ehrExtract,
+                        (Patient) new Patient().setId(PATIENT_ID),
+                        List.of(),
+                        PRACTISE_CODE));
 
-        assertAll(
-            () -> assertThat(planMedicationRequests).as("Plans").hasSize(1),
-            () -> assertThat(orderMedicationRequests).as("Orders").hasSize(2)
-        );
+        verify(medicationMapperContext).reset();
     }
 
     @Test
-    void When_MultipleOrdersAreBasedOnOneAcutePlan_Expect_AdditionalPlanCreated() {
-        setupMultipleOrdersToOnePlanStubs(ACUTE_PRESCRIPTION_EXTENSION);
-        var ehrExtract = unmarshallEhrExtract(
-            "ehrExtract_MultipleSupplyPrescribeInFulfilmentOfSingleAcuteSupplyAuthorise.xml"
-        );
-
-        var resources = medicationRequestMapper
-            .mapResources(ehrExtract, (Patient) new Patient().setId(PATIENT_ID), List.of(), PRACTISE_CODE
-        );
-
-        var planMedicationRequests = resources.stream()
-            .filter(MedicationRequest.class::isInstance)
-            .map(MedicationRequest.class::cast)
-            .filter(medicationRequest -> MedicationRequestIntent.PLAN.equals(medicationRequest.getIntent()));
+    public void When_MedicationRequestMapperCompletesSuccessfully_Expect_MedicationMapperContextToBeReset() {
+        var ehrExtract = unmarshallEhrExtract("ehrExtract1.xml");
 
-        var orderMedicationRequests = resources.stream()
-            .filter(MedicationRequest.class::isInstance)
-            .map(MedicationRequest.class::cast)
-            .filter(medicationRequest -> MedicationRequestIntent.ORDER.equals(medicationRequest.getIntent()))
-            .toList();
+        medicationRequestMapper
+                .mapResources(ehrExtract, (Patient) new Patient().setId(PATIENT_ID), List.of(), PRACTISE_CODE);
 
-        assertAll(
-            () -> assertThat(planMedicationRequests).as("Plans").hasSize(2),
-            () -> assertThat(orderMedicationRequests).as("Orders").hasSize(2)
-        );
+        verify(medicationMapperContext).reset();
     }
 
-    @Test
-    void When_MultipleOrdersAreBasedOnOneAcutePlan_Expect_TheGeneratedPlanHasIdAndIdentityUpdatedToGeneratedId() {
-        setupMultipleOrdersToOnePlanStubs(ACUTE_PRESCRIPTION_EXTENSION);
-        var ehrExtract = unmarshallEhrExtract(
-            "ehrExtract_MultipleSupplyPrescribeInFulfilmentOfSingleAcuteSupplyAuthorise.xml"
-        );
-
-        var resources = medicationRequestMapper
-            .mapResources(ehrExtract, (Patient) new Patient().setId(PATIENT_ID), List.of(), PRACTISE_CODE
-            );
+    @Nested()
+    @DisplayName("WhenTwoEhrSupplyPrescribeReferenceOneRepeatEhrSupplyAuthorise")
+    class TwoEhrSupplyPrescribeReferenceOneRepeatEhrSupplyAuthorise {
```

Comment:
Given that the display is only 4 characters different to the class name, what was the thinking for using DisplayName?